bgpd/bgp_vty.c

```diff
@@ -11489,7 +11490,9 @@ DEFUN (bgp_redistribute_ipv4_rmap,
 	}
 
 	red = bgp_redist_add(bgp, AFI_IP, type, 0);
-	bgp_redistribute_rmap_set(red, argv[idx_word]->arg);
+	changed = bgp_redistribute_rmap_set(red, argv[idx_word]->arg);
+	if (changed)
+		bgp_redistribute_unreg(bgp, AFI_IP, type, 0);
 	return bgp_redistribute_set(bgp, AFI_IP, type, 0);
```

Comment:
Let's add a `bool changed` parameter to bgp_redistribute_set and move the `if (changed)` if statement to inside of this function and just pass in changed to it.  This way if we add a new redistirbute statement somewhere or redo something else this knowledge is kept.